### PR TITLE
#18 add port to overwritten Host value when it different than 80

### PR DESCRIPTION
**Pull request by parry84**
This PR contains a fix for Issue #18

### CODE REVIEW

File: rawhttp-core/src/main/java/rawhttp/core/RawHttp.java [hunk 1]
```diff
@@ -404,4 +404,11 @@ private RequestLine verifyHost(RequestLine requestLine, RawHttpHeaders.Builder h
         }
     }
 
+    private String composeHostPort(URI uri) {
```
Comment (renatoathaydes) [inline rawhttp-core/src/main/java/rawhttp/core/RawHttp.java]: This method can be static... also, it seems to be used solely to set the value of the host header, so I suggest changing its name to `hostHeaderValueFor(URI uri)`.

File: rawhttp-core/src/main/java/rawhttp/core/RawHttp.java [hunk 2]
```diff
@@ -404,4 +404,11 @@ private RequestLine verifyHost(RequestLine requestLine, RawHttpHeaders.Builder h
...
+        if ((uri.getPort() >= 0) && (uri.getPort() != 80)) {
```
Comment (renatoathaydes) [inline rawhttp-core/src/main/java/rawhttp/core/RawHttp.java]: Why special-case port 80 only? If this is a `https` URI, I would expect port 43 to be used by default... should that still be added to the host header?

Comment (renatoathaydes) [reply rawhttp-core/src/main/java/rawhttp/core/RawHttp.java]: The default for https is actually 443: https://tools.ietf.org/html/rfc2818#section-2.3
I suggest adding a special case for that also, after checking which scheme the uri has.

**Comment by parry84** [comment]
Unable to change source branch: creating new PR: https://github.com/renatoathaydes/rawhttp/pull/20

**Pull request closed without merging**